python/cudf/cudf/core/series.py

```diff
@@ -1236,21 +1236,17 @@ def where(self, cond, other=None, axis=None):
         3
         4
         """
-
+        if not pd.api.types.is_numeric_dtype(self.dtype):
+            raise TypeError(
+                "`Series.where` is only supported for numeric column types."
+            )
```

Comment:
Is it the libcudf function underneath `find_and_replace` that doesn't support strings?

Reply:
For string columns `find_and_replace` only works when `to_replace` is a scalar. For categorical columns the existing logic threw a dtype error and was not tested. (Though I think there might be a way to get that working by modifying the logic). In general the `find_and_replace` approach is probably not the ideal approach for handling `series.where` under the hood (More details here: #3304). Ideally this could be replaced by a boolean setitem or use the newer `copy_if_else` libcudf method